mdata/cache/if.go

```diff
@@ -6,6 +6,7 @@ import (
 
 type Cache interface {
 	Add(string, uint32, chunk.IterGen)
+	CacheIfHot(string, uint32, *chunk.IterGen)
```

Comment:
why pointer to IterGen? this type is tiny and we can just copy it by value. semantically it also makes more sense to me, this implies we may make changes to the itergen.